tests/test_alertDataGenerator.py

```diff
@@ -0,0 +1,528 @@
+import unittest
+import os
+import numpy as np
+import tempfile
+import sqlite3
+import shutil
+import numbers
+import gc
+import lsst.utils.tests
+
+from lsst.utils import getPackageDir
+from lsst.sims.utils.CodeUtilities import sims_clean_up
+from lsst.sims.catalogs.decorators import register_method
+from lsst.sims.catalogs.db import CatalogDBObject
+from lsst.sims.catalogs.db import DBObject
+from lsst.sims.coordUtils import lsst_camera
+from lsst.sims.coordUtils import chipNameFromPupilCoordsLSST
+from lsst.sims.catUtils.utils import ObservationMetaDataGenerator
+from lsst.sims.catUtils.utils import AlertStellarVariabilityCatalog
+from lsst.sims.catUtils.utils import AlertDataGenerator
+from lsst.sims.catUtils.utils import StellarAlertDBObjMixin
+
+from lsst.sims.utils import applyProperMotion
+from lsst.sims.utils import ModifiedJulianDate
+from lsst.sims.utils import findHtmid
+from lsst.sims.utils import angularSeparation
+from lsst.sims.photUtils import Sed
+from lsst.sims.coordUtils import chipNameFromRaDecLSST
+from lsst.sims.coordUtils import pixelCoordsFromRaDecLSST
+from lsst.sims.photUtils import calcSNR_m5, BandpassDict
+from lsst.sims.photUtils import PhotometricParameters
+from lsst.sims.catUtils.mixins import CameraCoordsLSST
+from lsst.sims.catUtils.mixins import AstrometryStars
+from lsst.sims.catUtils.mixins import Variability
+from lsst.sims.catalogs.definitions import InstanceCatalog
+from lsst.sims.catalogs.decorators import compound, cached
+
+ROOT = os.path.abspath(os.path.dirname(__file__))
+
+
+def setup_module(module):
+    lsst.utils.tests.init()
+
+
+class AlertDataGeneratorTestCase(unittest.TestCase):
+
+    longMessage = True
+
+    @classmethod
+    def setUpClass(cls):
+        print('setting up %s' % sims_clean_up.targets)
+
+        # These represent the dimmest magnitudes at which objects
+        # are considered visible in each of the LSST filters
+        # (taken from Table 2 of the overview paper)
+        cls.obs_mag_cutoff = (23.68, 24.89, 24.43, 24.0, 24.45, 22.60)
```

Comment:
this is for my own edification only (probably a silly question too) -- why use cls.* instead of self.* ? Is this because it's required by the new multi-thread unit tests? (so you can share these variables across all instances of the class?) If you didn't - would the only problem here be that you'd have multiple connections to the database / multiple reads, etc.).

Reply:
My understanding was that `cls` and `self` are mechanically the same thing, it's just convention that, in `classmethod`s, you use `cls` so that it is clear that the member is associated with the class, rather than a specific instance of the class (and that, changing `cls.blah` in one instance will affect `blah` for all future instances).

In this particular case, I put the tuple in `setUpClass` because the setup was expensive and I only wanted it to run once for the whole script, rather than once for each test method (not that there's much difference between those scenarios in the post-`pytest` era).

Reply:
(I could be totally wrong about the distinction between `self` and `cls`, fyi)